source/image/ImageBuffer.cpp

```diff
@@ -153,13 +158,11 @@ void ImageBuffer::ShrinkToHalfSize()
 
 bool ImageBuffer::Read(const string &path, int frame)
 {
-	// First, make sure this is a JPG or PNG file.
-	if(path.length() < 4)
-		return false;
+	// First, make sure this is a supported file.
+	string extension = Format::LowerCase(path.substr(path.find_last_of('.') + 1));
```

Comment:
It might be worth looking into `std::filesystem::path` for this and similar operations.

Reply:
Yeah, we should be using that instead of strings everywhere. I just felt like that would touch too many things for this refactor.